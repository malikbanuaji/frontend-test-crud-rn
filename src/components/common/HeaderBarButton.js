import React from 'react';
import {StyleSheet, Text, TouchableNativeFeedback, View} from 'react-native';
import {Spacing, Typography, Mixins} from '../../styles';

export default function HeaderBarButton({
  text,
  onPress,
  style,
  textStyle,
  disabled,
}) {
  return (
    <View style={[styles.baseButton, disabled && styles.disabled, style]}>
      <TouchableNativeFeedback disabled={disabled} onPress={onPress}>
        <View style={styles.actionButton}>
          <Text style={[styles.actionButtonText, textStyle]}>{text}</Text>
        </View>
      </TouchableNativeFeedback>
    </View>
  );
}

const styles = StyleSheet.create({
  actionButtonText: {
    ...Typography.headerButtonText,
  },
  actionButton: {
    padding: Spacing.smaller,
  },
  baseButton: {
    ...Mixins.button,
    ...Mixins.buttonBorder,
  },
  disabled: {
    opacity: 0.5,
  },
});
