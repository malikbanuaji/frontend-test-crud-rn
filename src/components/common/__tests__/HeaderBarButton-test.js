import React from 'react';
import renderer from 'react-test-renderer';
import HeaderBarButton from '../HeaderBarButton';

test('renders correctly', () => {
  const tree = renderer.create(<HeaderBarButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
