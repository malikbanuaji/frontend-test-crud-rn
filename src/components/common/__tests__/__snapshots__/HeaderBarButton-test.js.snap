// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#808080",
        "borderRadius": 4,
        "borderWidth": 1,
        "overflow": "hidden",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "padding": 8,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Rubik-Medium",
            "fontSize": 14,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
